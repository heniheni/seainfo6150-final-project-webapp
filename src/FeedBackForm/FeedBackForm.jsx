import React, { useState } from "react";
import styles from "./FeedBackForm.module.css";

const Form = () => {
  const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    console.log("DATA", data.get("foo"));
    setSubmittedForm(data);
  }

  const [nameError, setNameError] = useState();
  const validateName = (e) => {
    var name = e.target.value;
    if (!name.match(/^[a-zA-Z ]*$/)) {
      setNameError("Enter Alphabet Character Only");
    } else {
      setNameError("");
    }
  };

  const [emailError, setEmailError] = useState();
  const validateEmail = (e) => {
    var email = e.target.value;
    var pattern = new RegExp(
      /^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i
    );
    if (!email.match(pattern)) {
      setEmailError("Enter Valid Email E.g. userXXX@outlook.com");
    } else {
      setEmailError("");
    }
  };

  const [mobileNumberError, setMobileNumberError] = useState();
  const validateMobileNumber = (e) => {
    var mobile = e.target.value;
    if (!mobile.match(/^[0-9]{10}$/)) {
      setMobileNumberError("Enter valid Phone Number");
    } else {
      setMobileNumberError("");
    }
  };

  return (
    <div>
      {submittedForm ? (
        <div className={styles.inputDataContainer}>
          <h1>Thank You !! We got your Feedback</h1>
          <div className={styles.fields}>
            <b>Name: </b> {submittedForm.get("name")}
          </div>
          
          <div className={styles.fields}>
            <b>I have a Feedback About: </b>{submittedForm.get("myDropdown")}
          </div>

          <div className={styles.fields}>
            <b>Did you Achieve your Goal? </b>{submittedForm.get("radio")}
          </div>

          <div className={styles.fields}>
            <b>E-mail: </b>{submittedForm.get("email")}
          </div>

          <div className={styles.fields}>
            <b>Phone No: </b>{submittedForm.get("mobile")}
          </div>

          <div className={styles.fields}>
            <b>City: </b>{submittedForm.get("myDropdownCity")}
          </div>

          <div className={styles.fields}>
            <b>Zipcode: </b>{submittedForm.get("zipcode")}
          </div>

          <div className={styles.fields}>
            <b>Comments: </b>{submittedForm.get("comments")}
          </div>

          <div className={styles.fields}>
            <b>Subscribe for Daily Update: </b>{submittedForm.get("checkbox")}
          </div> 
        </div>
      ) : (
        <form onSubmit={onSubmit}>
          <div className={styles.formContainer}>
            <h1 className={styles.formName}>Feedback Form</h1>
            <p className={styles.subText}>
              Please fill in this form to give your Feedback.
              <br />
              <br />
              <i>Fields' marked with * are mandatory </i>
            </p>
            <br />
            <hr />
            <label>I have a Feedback About: </label>
            <select name="myDropdown">
              <option value="Website Issue">Website Issue</option>
              <option value="My Experience">My Experience</option>
              <option value="Improvement">Improvement</option>
            </select>
            <br />
            <label>Did you Achieve your Goal? </label>
            <label>Yes</label>
            <input type="radio" name="radio" id="foo-yes" value="yes" />
            <label>No</label>
            <input type="radio" name="radio" id="foo-no" value="no" />
            <label>Maybe</label>
            <input type="radio" name="radio" id="foo-maybe" value="maybe" />
            <br />
            <label htmlFor="myTextId">*Name: </label>{" "}
            <label className={styles.stylingError}>{nameError}</label>
            <br />
            <br />
            <input
              type="text"
              id="name"
              name="name"
              required
              onChange={(e) => validateName(e)}
            />
            <br />
            <label>*E-mail: </label>{" "}
            <label className={styles.stylingError}>{emailError}</label>
            <br />
            <br />
            <input
              type="text"
              id="userEmail"
              name="email"
              required
              onChange={(e) => validateEmail(e)}
            />
            <label>*Phone No: </label>{" "}
            <label className={styles.stylingError}>{mobileNumberError}</label>
            <br />
            <input
              type="text"
              id="mobile"
              name="mobile"
              required
              onChange={(e) => validateMobileNumber(e)}
            />
            <br />
            <label htmlFor="myDropdownId">City: </label>
            <select
              className={styles.dropDown}
              name="myDropdownCity"
              id="myDropdownId"
            >
              <option value="Chennai">Chennai</option>
              <option value="Mumbai">Mumbai</option>
              <option value="Kolkata">Kolkata</option>
              <option value="Bengaluru">Bengaluru</option>
              <option value="New Delhi">New Delhi</option>
              <option value="Mohali">Mohali</option>
              <option value="Jaipur">Jaipur</option>
              <option value="Hyderbad">Hyderbad</option>
            </select>
            <br />
            <br />
            <label>Zip Code: </label>
            <br />
            <input type="text" name="zipcode" id="zipcode" />
            <br />
            <label>Comments: </label>
            <br />
            <input type="text" name="comments" id="comments" />
            <br />
            <label>Subscribe for Daily Update: </label>
            <label>Yes</label>
            <input type="checkbox" name="checkbox" id="yes-yes" value="yes" />
            <label>No</label>
            <input type="checkbox" name="checkbox" id="no-no" value="no" />
            <br />
            <input
              className={styles.submitButton}
              type="submit"
              value="Submit"
            />
          </div>
        </form>
      )}
    </div>
  );
};

export default Form;
