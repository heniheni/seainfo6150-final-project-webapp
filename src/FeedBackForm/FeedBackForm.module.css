.inputDataContainer{

    display: block;
    box-shadow: 4px 3px 8px 1px grey;
    width: 60%;
    padding: 20px;
    margin-left: auto;
    margin-right: auto;
    margin-top: 10%;
    margin-bottom: 20%;

 
}
.stylingBreadCrumb{
  text-decoration: none;
}
.fields{
  padding: 10px;
 
}
.formContainer{
    padding: 10px;
}
.stylingError{
  color: red;
  font-style: italic;
  font-weight: bold;
}

label{
    font-weight: bold;
}

.formName{
    text-align: center;
}

.subText{
    text-align: center;
}
::placeholder{
  color: darkgray;
  filter: opacity(0.5);
  padding-left: 10px;
}

input:focus::placeholder {
  color: transparent;
}

hr {
    border: 1px solid #f1f1f1;
    margin-bottom: 15px;
}

  input[type="checkbox"]{
    margin: 20px;

    background: #f1f1f1;
  }

  input[type="radio"]{
    margin: 20px;
    background: #f1f1f1;
  }

.dropDown{
    background: #f1f1f1;
}
input[type="text"]{
    width: 100%;
    height: 40px;
    /* padding: 15px; */
    margin: 5px 0 22px 0;
    display: inline-block;
    border: none;
    background: #f1f1f1;
    margin-right: 15px;
    border-radius: 10px;   
}
input:focus{
 outline: none;
  border: none;
  box-shadow: #19398a 0px 1px 2px, rgba(0, 0, 0, 0.07) 0px 2px 4px, rgba(0, 0, 0, 0.07) 0px 4px 8px, rgba(0, 0, 0, 0.07) 0px 8px 16px, rgba(0, 0, 0, 0.07) 0px 16px 32px, rgba(0, 0, 0, 0.07) 0px 32px 64px;
}

.submitButton {
    background-color: #19398a;
    color: white;
    /* padding: 16px 20px;  */
    border: none;
    cursor: pointer;
    width: 100%;
    
    height: 40px;
    
    margin-right: 15px;
  }

  .submitButton:hover {
    opacity:1;
  }