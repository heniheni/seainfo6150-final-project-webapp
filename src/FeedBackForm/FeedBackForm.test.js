import React from "react";
import FeedBackForm from "./FeedBackForm";


describe("FeedBackForm tests", () => {
    const dummy = () => " ";
    it("renders correctly", () => {
      const { container } = render(<FeedBackForm Name="tryTest" E-mail="test@gmail.com" Phone No="2004001110" City="Surat" Zip Code="98121" Comments="Testing"  onClick={dummy}/>);
      expect(container).toMatchSnapshot();
    });
  });