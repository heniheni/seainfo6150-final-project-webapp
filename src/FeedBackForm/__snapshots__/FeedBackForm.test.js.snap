// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedBackForm tests renders correctly 1`] = `
<div>
  <div>
    <form>
      <div
        class="formContainer"
      >
        <h1
          class="formName"
        >
          Feedback Form
        </h1>
        <p
          class="subText"
        >
          Please fill in this form to give your Feedback.
          <br />
          <br />
          <i>
            Fields' marked with * are mandatory 
          </i>
        </p>
        <br />
        <hr />
        <label>
          I have a Feedback About: 
        </label>
        <select
          name="myDropdown"
        >
          <option
            value="Website Issue"
          >
            Website Issue
          </option>
          <option
            value="My Experience"
          >
            My Experience
          </option>
          <option
            value="Improvement"
          >
            Improvement
          </option>
        </select>
        <br />
        <label>
          Did you Achieve your Goal? 
        </label>
        <label>
          Yes
        </label>
        <input
          id="foo-yes"
          name="radio"
          type="radio"
          value="yes"
        />
        <label>
          No
        </label>
        <input
          id="foo-no"
          name="radio"
          type="radio"
          value="no"
        />
        <label>
          Maybe
        </label>
        <input
          id="foo-maybe"
          name="radio"
          type="radio"
          value="maybe"
        />
        <br />
        <label
          for="myTextId"
        >
          *Name: 
        </label>
         
        <label
          class="stylingError"
        />
        <br />
        <br />
        <input
          id="name"
          name="name"
          required=""
          type="text"
        />
        <br />
        <label>
          *E-mail: 
        </label>
         
        <label
          class="stylingError"
        />
        <br />
        <br />
        <input
          id="userEmail"
          name="email"
          required=""
          type="text"
        />
        <label>
          *Phone No: 
        </label>
         
        <label
          class="stylingError"
        />
        <br />
        <input
          id="mobile"
          name="mobile"
          required=""
          type="text"
        />
        <br />
        <label
          for="myDropdownId"
        >
          City: 
        </label>
        <select
          class="dropDown"
          id="myDropdownId"
          name="myDropdownCity"
        >
          <option
            value="Chennai"
          >
            Chennai
          </option>
          <option
            value="Mumbai"
          >
            Mumbai
          </option>
          <option
            value="Kolkata"
          >
            Kolkata
          </option>
          <option
            value="Bengaluru"
          >
            Bengaluru
          </option>
          <option
            value="New Delhi"
          >
            New Delhi
          </option>
          <option
            value="Mohali"
          >
            Mohali
          </option>
          <option
            value="Jaipur"
          >
            Jaipur
          </option>
          <option
            value="Hyderbad"
          >
            Hyderbad
          </option>
        </select>
        <br />
        <br />
        <label>
          Zip Code: 
        </label>
        <br />
        <input
          id="zipcode"
          name="zipcode"
          type="text"
        />
        <br />
        <label>
          Comments: 
        </label>
        <br />
        <input
          id="comments"
          name="comments"
          type="text"
        />
        <br />
        <label>
          Subscribe for Daily Update: 
        </label>
        <label>
          Yes
        </label>
        <input
          id="yes-yes"
          name="checkbox"
          type="checkbox"
          value="yes"
        />
        <label>
          No
        </label>
        <input
          id="no-no"
          name="checkbox"
          type="checkbox"
          value="no"
        />
        <br />
        <input
          class="submitButton"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
</div>
`;
